Octokit.Reactive/Clients/ObservableCommitStatusClient.cs

```diff
@@ -28,10 +28,34 @@ public ObservableCommitStatusClient(IGitHubClient client)
         /// <returns></returns>
         public IObservable<CommitStatus> GetAll(string owner, string name, string reference)
         {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(name, "name");
+            Ensure.ArgumentNotNullOrEmptyString(reference, "reference");
```

Comment:
Sorry, I should clarify this before you go and fix it.

 - The line below is very similar to the actual implementation inside the new `GetAll` method you've defined. So rather than call `_connection.GetAndFlattenPages<T>(...)` here, why not just call `GetAll(owner, name, reference, ApiOptions.None);` here?

 - We've been discussing what to do with the `Ensure` usages in the case where we call a different overload. We've settled on keeping these around, even if they are duplicated checks. So don't clean them up here.

Make sense?